core/include/detray/geometry/surfaces_finder.hpp

```diff
@@ -0,0 +1,137 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "detray/definitions/detail/accessor.hpp"
+#include "detray/grids/axis.hpp"
+#include "detray/grids/grid2.hpp"
+#include "detray/grids/populator.hpp"
+#include "detray/grids/serializer2.hpp"
+#include "detray/tools/grid_array_helper.hpp"
+
+#pragma once
+
+namespace detray {
+
+template <std::size_t N,
+          template <typename, unsigned int> class array_type = darray,
+          template <typename...> class tuple_type = dtuple,
+          template <typename...> class vector_type = dvector,
+          template <typename...> class jagged_vector_type = djagged_vector>
+struct surfaces_finder {
+    public:
+    surfaces_finder() = delete;
+
+    using surfaces_regular_circular_grid =
```

Comment:
Not sure this should be here? Even if we can only support one global grid type currently, the detector defines what that is, right?

Reply:
It's for `surfaces_finder_data` and `surfaces_finder_view` If I get it right.

Reply:
I think, that we might see more than just one volume geometry (at least there are multiple in Acts), so I would be a bit reluctant to fix everything to ```regular_circular_grid```s, even though the detector can currently only handle that. What do you think?

Reply:
Please tell me If I get your statement wrong.
If I understand correctly, we may need to do `tuple<array[grid_type1], array[grid_type2], ...>` <<< is it right?

Reply:
Or we can just list down
```c++
array[grid_type1] surfaces_finders1;
array[grid_type2] surfaces_finders2;
...
```

Reply:
BTW, I guess we only need `regular_circular` or `regular_regular` grid. There is no point of using irregular type for surface finder.

Reply:
Yes, I think that should be enough. Anyway, I mainly wanted to give you a heads up, we can implement this in a later PR